src/MySqlConnector/Protocol/Serialization/ByteBufferWriter.cs

```diff
@@ -259,6 +259,27 @@ public static void WriteLengthEncodedString(this ByteBufferWriter writer, ReadOn
 		}
 #endif
 
+		public static void WriteLengthEncodedString(this ByteBufferWriter writer, StringBuilder builder)
+		{
+#if !NET45 && !NET461 && !NET471 && !NETSTANDARD1_3 && !NETSTANDARD2_0 && !NETSTANDARD2_1 && !NETCOREAPP2_1
+			var totalLength = 0;
+
+			foreach (var chunk in builder.GetChunks())
+			{
+				totalLength += Encoding.UTF8.GetByteCount(chunk.Span);
```

Comment:
I suspect this will fail if the just happens to end with a leading surrogate code unit, and the next chunk begins with the matching trailing surrogate.

This would probably need to use an `Encoder` that tracks state, and can account for a the two code units being split across two chunks.

Reply:
this was fixed in the further commits, thus I'm marking this as a resolved